### ISSUE TITLE
Python 3.5.2 compatibility broken between 0.56.0 and 0.56.1

### ISSUE THREAD

**Issue by karliss** [mesonbuild/meson#8161]
**Describe the bug**
Version 0.56.1 doesn't work with python 3.5.2 can't even run `meson --version`.  0.56.0 worked.  From all the information I could find python requirement is increased to 3.6 in master but it wasn't supposed happen in 0.56.x.

**To Reproduce**
 * `pip3 install --user meson==0.56.1`
 * run `meson --version`

```
  File "/home/test/.local/bin/meson", line 7, in <module>
    from mesonbuild.mesonmain import main
  File "/home/test/.local/lib/python3.5/site-packages/mesonbuild/mesonmain.py", line 25, in <module>
    from . import mconf, mdist, minit, minstall, mintro, msetup, mtest, rewriter, msubprojects, munstable_coredata, mcompile
  File "/home/test/.local/lib/python3.5/site-packages/mesonbuild/mconf.py", line 16, in <module>
    from . import coredata, environment, mesonlib, build, mintro, mlog
  File "/home/test/.local/lib/python3.5/site-packages/mesonbuild/build.py", line 26, in <module>
    from . import dependencies
  File "/home/test/.local/lib/python3.5/site-packages/mesonbuild/dependencies/__init__.py", line 15, in <module>
    from .boost import BoostDependency
  File "/home/test/.local/lib/python3.5/site-packages/mesonbuild/dependencies/boost.py", line 26, in <module>
    from .base import DependencyException, ExternalDependency, PkgConfigDependency
  File "/home/test/.local/lib/python3.5/site-packages/mesonbuild/dependencies/base.py", line 37, in <module>
    from ..cmake import CMakeExecutor, CMakeTraceParser, CMakeException, CMakeToolchain, CMakeExecScope, check_cmake_args
  File "/home/test/.local/lib/python3.5/site-packages/mesonbuild/cmake/__init__.py", line 42, in <module>
    from .interpreter import CMakeInterpreter
  File "/home/test/.local/lib/python3.5/site-packages/mesonbuild/cmake/interpreter.py", line 528
    res: T.List[str] = []
       ^
```

**Expected behavior**
Python version requirements aren't increased in minor version update.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? : N/A
* Ubuntu 16.04
* 3.5.2
* what `meson --version` -> crashes when running `meson --version`, but the installed version is 0.56.1

**Comment by jpakkane** [comment]
Can you test if #8164 fixes it for you?

**Comment by karliss** [comment]
Yes, that seemed to help.

**Comment by nirbheek** [comment]
This is fixed in the 0.56 branch, and we have dropped Python 3.5 support in master. We will release 0.56.2 with the fix soon.
